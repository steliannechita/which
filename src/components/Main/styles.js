import styled, { createGlobalStyle } from "styled-components";
import { media } from "../../utils";

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #a1e4b2;
    height: auto;
    margin: 0;
    font-family: "Roboto", sans-serif;
    display:flex;
    justify-content:center;
    align-items:center;
  }
  h1,p{margin:0}
`;

export const MainWrapper = styled.div`
  h1 {
    font-style: italic;
    color: #794616;
    width: 100%;
    text-align: center;
    padding: 10px;
    ${media.mobile} {
      font-size: 24px;
    }
  }
  p {
    width: 100%;
    margin: 0;
    text-align: center;
    margin-bottom: 10px;
    &.remember {
      font-style: italic;
      color: #794616;
    }
  }
  max-width: 700px;
  display: flex;
  flex-flow: wrap;
  justify-content: center;
  align-items: center;
  border: 2px solid #794616;
  padding: 0 5px 20px 5px;
  margin: 15px;
`;
