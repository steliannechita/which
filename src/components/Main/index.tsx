import React, { useState } from "react";
import { NumberBox } from "../.";
import { MainWrapper, GlobalStyle } from "./styles";
import { initialArray, NumberElement } from "../../utils";

const Main = () => {
  const [numbersArray, setNumbersArray] = useState(initialArray);

  const handleClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
    const { value } = e.currentTarget;
    const valueAsNumber = +value;
    const newNumbersArray: NumberElement[] = numbersArray.map(
      (number: NumberElement) => {
        if (number.numberValue % valueAsNumber === 0) {
          if (number.numberValue === valueAsNumber) {
            return { ...number, isMultiple: true, isSelected: true };
          }
          return { ...number, isMultiple: true, isSelected: false };
        }
        return { ...number, isMultiple: false, isSelected: false };
      }
    );
    setNumbersArray(newNumbersArray);
  };

  return (
    <>
      <MainWrapper>
        <h1>TABLE OF MULTIPLES</h1>
        <p>Select a number from the grid to see its multiples </p>
        <p className="remember">
          * Also remember, every number is always a multiple of itself *
        </p>

        {numbersArray.map((item) => (
          <NumberBox
            key={item.numberValue}
            numberValue={item.numberValue}
            isMultiple={item.isMultiple}
            isSelected={item.isSelected}
            handleClick={handleClick}
          />
        ))}
      </MainWrapper>
      <GlobalStyle />
    </>
  );
};

export default Main;
