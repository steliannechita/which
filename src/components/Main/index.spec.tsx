import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Main from "./";

describe("Main", () => {
  it("should render successfully", () => {
    const { baseElement } = render(<Main />);
    expect(baseElement).toMatchSnapshot();
  });
  it("should correctly select a number when clicked", () => {
    const { getByText } = render(<Main />);
    fireEvent.click(getByText("1"));
    expect(getByText("1")).toHaveClass("with-border");
  });
  it("should correctly highlight multiples of a number", () => {
    const { getByText } = render(<Main />);
    fireEvent.click(getByText("50"));
    expect(getByText("50")).toHaveClass("highlight");
    expect(getByText("100")).toHaveClass("highlight");
  });
});
