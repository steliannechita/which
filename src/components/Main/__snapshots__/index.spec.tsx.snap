// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render successfully 1`] = `
<body>
  <div>
    <div
      class="sc-gsTCUz bLAJpv"
    >
      <h1>
        TABLE OF MULTIPLES
      </h1>
      <p>
        Select a number from the grid to see its multiples 
      </p>
      <p
        class="remember"
      >
        * Also remember, every number is always a multiple of itself *
      </p>
      <button
        class="sc-bdfBwQ gErxWH"
        value="1"
      >
        1
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="2"
      >
        2
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="3"
      >
        3
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="4"
      >
        4
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="5"
      >
        5
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="6"
      >
        6
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="7"
      >
        7
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="8"
      >
        8
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="9"
      >
        9
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="10"
      >
        10
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="11"
      >
        11
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="12"
      >
        12
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="13"
      >
        13
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="14"
      >
        14
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="15"
      >
        15
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="16"
      >
        16
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="17"
      >
        17
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="18"
      >
        18
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="19"
      >
        19
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="20"
      >
        20
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="21"
      >
        21
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="22"
      >
        22
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="23"
      >
        23
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="24"
      >
        24
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="25"
      >
        25
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="26"
      >
        26
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="27"
      >
        27
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="28"
      >
        28
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="29"
      >
        29
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="30"
      >
        30
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="31"
      >
        31
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="32"
      >
        32
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="33"
      >
        33
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="34"
      >
        34
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="35"
      >
        35
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="36"
      >
        36
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="37"
      >
        37
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="38"
      >
        38
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="39"
      >
        39
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="40"
      >
        40
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="41"
      >
        41
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="42"
      >
        42
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="43"
      >
        43
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="44"
      >
        44
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="45"
      >
        45
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="46"
      >
        46
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="47"
      >
        47
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="48"
      >
        48
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="49"
      >
        49
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="50"
      >
        50
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="51"
      >
        51
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="52"
      >
        52
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="53"
      >
        53
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="54"
      >
        54
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="55"
      >
        55
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="56"
      >
        56
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="57"
      >
        57
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="58"
      >
        58
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="59"
      >
        59
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="60"
      >
        60
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="61"
      >
        61
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="62"
      >
        62
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="63"
      >
        63
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="64"
      >
        64
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="65"
      >
        65
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="66"
      >
        66
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="67"
      >
        67
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="68"
      >
        68
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="69"
      >
        69
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="70"
      >
        70
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="71"
      >
        71
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="72"
      >
        72
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="73"
      >
        73
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="74"
      >
        74
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="75"
      >
        75
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="76"
      >
        76
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="77"
      >
        77
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="78"
      >
        78
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="79"
      >
        79
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="80"
      >
        80
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="81"
      >
        81
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="82"
      >
        82
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="83"
      >
        83
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="84"
      >
        84
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="85"
      >
        85
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="86"
      >
        86
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="87"
      >
        87
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="88"
      >
        88
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="89"
      >
        89
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="90"
      >
        90
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="91"
      >
        91
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="92"
      >
        92
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="93"
      >
        93
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="94"
      >
        94
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="95"
      >
        95
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="96"
      >
        96
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="97"
      >
        97
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="98"
      >
        98
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="99"
      >
        99
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="100"
      >
        100
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="101"
      >
        101
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="102"
      >
        102
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="103"
      >
        103
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="104"
      >
        104
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="105"
      >
        105
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="106"
      >
        106
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="107"
      >
        107
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="108"
      >
        108
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="109"
      >
        109
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="110"
      >
        110
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="111"
      >
        111
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="112"
      >
        112
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="113"
      >
        113
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="114"
      >
        114
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="115"
      >
        115
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="116"
      >
        116
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="117"
      >
        117
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="118"
      >
        118
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="119"
      >
        119
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="120"
      >
        120
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="121"
      >
        121
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="122"
      >
        122
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="123"
      >
        123
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="124"
      >
        124
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="125"
      >
        125
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="126"
      >
        126
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="127"
      >
        127
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="128"
      >
        128
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="129"
      >
        129
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="130"
      >
        130
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="131"
      >
        131
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="132"
      >
        132
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="133"
      >
        133
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="134"
      >
        134
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="135"
      >
        135
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="136"
      >
        136
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="137"
      >
        137
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="138"
      >
        138
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="139"
      >
        139
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="140"
      >
        140
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="141"
      >
        141
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="142"
      >
        142
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="143"
      >
        143
      </button>
      <button
        class="sc-bdfBwQ gErxWH"
        value="144"
      >
        144
      </button>
    </div>
  </div>
</body>
`;
